docs/cookbook/dynamic-configuration/02_Templates/01_Looping.md

```diff
@@ -0,0 +1,42 @@
+# 01 Basics of Templating
+
+The templating is based on the excellent [gomplate](https://docs.gomplate.ca/) library. Currently using a custom fork to allow loading gomplate as a library in addition to some new commands. This will NOT try to cover the full range of gomplate, would recommend reading the documentation for full knowledge. 
+
+`docker run -v ${PWD}/:/etc/grafana grafana/agentctl:latest template-parse /etc/grafana/01_config.yml`
+
+## Looping
+
+[agent-1.yml](./01_assets/agent-1.yml)
+
+```yaml
+server:
+  http_listen_port: 12345
+  log_level: debug
+metrics:
+  wal_directory: /tmp/grafana-agent-normal
+  global:
+    scrape_interval: 60s
+    remote_write:
+      - url: https://prometheus-us-central1.grafana.net/api/prom/push
+        basic_auth:
+          username: 12345
+          password: secretpassword
+  configs:
+    - name: default
+  {{ range slice "apple" "banana" "pear" }}
+    - name: {{ . }}
```

Comment:
Nice! I think we should make it clear what happens if you have metrics instances defined in the base config file and also in templates. (IMO, it should be valid to have both and just append the arrays, if that's not already what happens)

Reply:
It is appended, but I will add more verbiage on what is additive vs what is replaced.